metadata-ingestion/src/datahub/ingestion/source/source_registry.py

```diff
@@ -1,94 +1,5 @@
 from datahub.ingestion.api.registry import Registry
 from datahub.ingestion.api.source import Source
 
-from .mce_file import MetadataFileSource
-
 source_registry = Registry[Source]()
```

Comment:
It would be amazing if the registry could use the type annotations to do some checking at runtime - that way the `load()` method wouldn't need the redundant parameter.

Probably not in scope for this PR though 🙂

Reply:
Using the types at runtime would be amazing, but seems to be tricky. I think there are libraries that can do this (e.g. `typing_inspect`), but I agree that's probably outside the scope for this